src/ogr/feature.jl

```diff
@@ -402,13 +455,33 @@ const _FETCHFIELD = Dict{OGRFieldType,Function}(
     OFTInteger64List => asint64list,
 )
 
+"""
+    getfield(feature, i)
+
+When the field is unset, it will return `nothing`. When the field is set but
+null, it will return `missing`.
+
+### References
+* https://gdal.org/development/rfc/rfc53_ogr_notnull_default.html
+* https://gdal.org/development/rfc/rfc67_nullfieldvalues.html
+"""
 function getfield(feature::Feature, i::Integer)
-    return if isfieldset(feature, i)
-        _fieldtype = gettype(getfielddefn(feature, i))
-        _fetchfield = get(_FETCHFIELD, _fieldtype, getdefault)
-        _fetchfield(feature, i)
+    return if !isfieldset(feature, i)
+        nothing
+    elseif isfieldnull(feature, i)
+        missing
     else
-        getdefault(feature, i)
+        _fieldtype = getfieldtype(getfielddefn(feature, i))
+        try
+            _fetchfield = _FETCHFIELD[_fieldtype]
+            _fetchfield(feature, i)
+        catch e
+            if e isa KeyError
+                error("$_fieldtype not implemented in getfield, please report an issue to https://github.com/yeesian/ArchGDAL.jl/issues")
```

Comment:
This is helpful message, but I wonder if a try-catch in such an often called method might slow it down? The error without try catch will be worse, but not terrible.

Reply:
Yeah I was thinking of [turning the enums into types so that we can dispatch on them](https://discourse.julialang.org/t/style-recommendation-for-enum-as-type/46078/5), but that one commit grew and grew in scope, so I ended up keeping it out of this PR, with the rationale that performance optimizations can be pursued down the line once we've nailed the API.

Reply:
Ok indeed it would only be a performance optimization, which could be done later.

Reply:
Opened https://github.com/yeesian/ArchGDAL.jl/issues/246 to track it.